// [[Reason: Invalid computed property can only be detected by TSC]] ////

//// [tests/cases/conformance/es6/computedProperties/computedPropertyNames13_ES5.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -4,20 +4,23 @@
 declare var s: string;
 declare var n: number;
 declare var a: any;
 declare class C {
+    [s](): invalid;
+    [n](): invalid;
     static [""](): invalid;
     [0](): invalid;
+    [a](): invalid;
     [`hello bye`](): invalid;
 }
 
 /// [Errors] ////
 
-computedPropertyNames13_ES5.ts(5,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
-computedPropertyNames13_ES5.ts(6,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
+computedPropertyNames13_ES5.ts(5,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
+computedPropertyNames13_ES5.ts(6,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
 computedPropertyNames13_ES5.ts(10,12): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
 computedPropertyNames13_ES5.ts(11,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
-computedPropertyNames13_ES5.ts(12,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
+computedPropertyNames13_ES5.ts(12,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
 computedPropertyNames13_ES5.ts(14,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
 
 
 ==== computedPropertyNames13_ES5.ts (6 errors) ====
@@ -26,12 +29,14 @@
     var a: any;
     class C {
         [s]() {}
         ~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
+!!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
+!!! related TS9034 computedPropertyNames13_ES5.ts:5:5: Add a return type to the method
         [n]() { }
         ~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
+!!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
+!!! related TS9034 computedPropertyNames13_ES5.ts:6:5: Add a return type to the method
         static [s + s]() { }
         [s + n]() { }
         [+s]() { }
         static [""]() { }
@@ -43,9 +48,10 @@
 !!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
 !!! related TS9034 computedPropertyNames13_ES5.ts:11:5: Add a return type to the method
         [a]() { }
         ~~~
-!!! error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations
+!!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
+!!! related TS9034 computedPropertyNames13_ES5.ts:12:5: Add a return type to the method
         static [<any>true]() { }
         [`hello bye`]() { }
         ~~~~~~~~~~~~~
 !!! error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations
